### PR TITLE
Some Fixes, Feedback and UI improvements

### PR DESCRIPTION
**Pull request by Linux4Good**
HI!

**Ignore commits before the last merge since my branch was out of date**

Through the issues list I found a few things to work with:

- [x] Update some packages
- [x] Files not removed when canceling installation #288 
- [x] Add Warning Proton Message
- [x] Add Error when Python version is lower than 3.8
- [x] Add Error message when Credentials have expired #306 
- [x] Improve Refresh Library Performance
- [x] Added Weblate Link to attract more translators 
- [x] Added Launch function to Protocol (using `heroic://launch/{appName}`)
- [x] Refactor Install/uninstall functions on the frontend.
- [x] Refactor Play Function on FE
- [x] Refactor Dialogs

### CODE REVIEW

**Comment by adityaruplaha** [comment]
Hi! Thanks for the contributions. Since this PR addresses multiple parts of the codebase at once, will it be possible to split them up so that they can be reviewed and merged one at a time? 

Ideally each checklist item should be a PR but clubbing a few together is fine. But this PR is really large, and we've had problems with things slipping through while reviewing large PRs before. 

Thanks again!

**Comment by flavioislima** [comment]
> Hi! Thanks for the contributions. Since this PR addresses multiple parts of the codebase at once, will it be possible to split them up so that they can be reviewed and merged one at a time? 
> 
> Ideally each checklist item should be a PR but clubbing a few together is fine. But this PR is really large, and we've had problems with things slipping through while reviewing large PRs before. 
> 
> Thanks again!

Thanks for the comment. Yes, I used to contribute here before but I was working on other stuff and now I'm on vacation so I can get back to this amazing project.

About the PRs, it's a lot of changes but not that much code changed in itself, most things removed and some tests that I need to fix.
Don't make me create separate PRs because will be crazy to split the changes now that they are here on github 😭

**Comment by Linux4Good** [comment]
> Hi! Thanks for the contributions. Since this PR addresses multiple parts of the codebase at once, will it be possible to split them up so that they can be reviewed and merged one at a time?
> 
> Ideally each checklist item should be a PR but clubbing a few together is fine. But this PR is really large, and we've had problems with things slipping through while reviewing large PRs before.
> 
> Thanks again!

Thanks for the Feedback @adityaruplaha 

I used to contribute to Heroic in the past but got busy with other stuff.
Now that I'm on vacation this month I'll be able to do it again for this amazing project :P

Sorry about the size of the PR but split it now will be complicated.
I think most of the things here we removed and not added so should not be too much code to see. :)

**Comment by flavioislima** [comment]
@Nocccer Take a look at `helpers.test `I'm also don't know whats happening, whats missing.
As far as I can see the function has the same logic, its just in another place. I don't know why the tests are complaining about the showOpenDialog even when using the default path.

**Review by Nocccer — CHANGES_REQUESTED** [review]

File: src/components/utils.ts
Comment (flavioislima) [reply src/components/utils.ts:83]: This whole file was removed since those functions reside in the library helper file.

File: src/helpers/library.ts
Comment (flavioislima) [reply src/helpers/library.ts:52]: No. This way is correct. 

This mean that if the game is to be installed on another path than the default and also if is not a game, so its a unreal asset, you should choose the folder always.

So it's one or the other.
Anither path or an asset.

**Review by flavioislima — DISMISSED** [review]
Looks good to me but the helpers tests will neednto be rewritten soon. 👍

**Comment by flavioislima** [comment]
I will still wait for @adityaruplaha to review it before merging.

File: public/locales/en/translation.json
Comment (adityaruplaha) [inline public/locales/en/translation.json:15]:
```suggestion
                "message": "Your credentials have expired, please logout and login again.",
```

Will need to be fixed in other files too.

File: src/components/utils.ts
Comment (Linux4Good) [reply src/components/utils.ts:83]: Yeah, I refactored it and put it in another file since now this function is used in multiple places.

File: public/locales/en/translation.json
Comment (adityaruplaha) [inline public/locales/en/translation.json:27]:
```suggestion
            "python": "Heroic requires Python 3.8 or newer to work."
```

File: public/locales/en/translation.json
Comment (adityaruplaha) [inline public/locales/en/translation.json:119]:
```suggestion
        "weblate": "Help translate Heroic."
``

File: src/helpers/library.ts
Comment (adityaruplaha) [inline src/helpers/library.ts:25]:
```suggestion
  // Fix removing files when canceling (again)
```
Sorry for being so pedantic.

File: src/screens/Settings/components/Tools/index.tsx
Comment (adityaruplaha) [inline src/screens/Settings/components/Tools/index.tsx:29]:
```suggestion
    const { path } = await ipcRenderer.invoke('openDialog', {
```

File: src/screens/Settings/index.css
Comment (adityaruplaha) [inline src/screens/Settings/index.css:152]: Missing trailing newline.

File: src/state/GlobalState.tsx
Comment (adityaruplaha) [inline src/state/GlobalState.tsx:257]: Makes sense, but why was the original logic deleted?

File: src/state/GlobalState.tsx
Comment (adityaruplaha) [inline src/state/GlobalState.tsx:209]: Missing `bool` argument? Or inferred `false`?

File: src/state/GlobalState.tsx
Comment (adityaruplaha) [inline src/state/GlobalState.tsx:258]: Why not use `ipcRenderer.on`?

**Review by adityaruplaha — CHANGES_REQUESTED** [review]

File: electron/protocol.ts
Comment (Linux4Good) [reply electron/protocol.ts:21]: What do you mean?

File: src/screens/Settings/index.css
Comment (Linux4Good) [reply src/screens/Settings/index.css:152]: Damn newlines!

File: src/state/GlobalState.tsx
Comment (Linux4Good) [reply src/state/GlobalState.tsx:257]: Dont need anymore since the error is being treated on the backend

File: src/state/GlobalState.tsx
Comment (Linux4Good) [reply src/state/GlobalState.tsx:209]: The argument is optional. So it will evaluate to false.

File: src/state/GlobalState.tsx
Comment (Linux4Good) [reply src/state/GlobalState.tsx:258]: because once will guarantee that the function won't launch the game twice. I'm also checking if the game have some status, to avoid other errors.

**Comment by Linux4Good** [comment]
@adityaruplaha I'll address the grammar issues but other than that looks good?

File: src/screens/Settings/components/Tools/index.tsx
Comment (Linux4Good) [reply src/screens/Settings/components/Tools/index.tsx:29]: good catch

File: src/helpers/library.ts
Comment (Linux4Good) [reply src/helpers/library.ts:25]: this was fixed, just forgot to remove the reminder

File: electron/protocol.ts
Comment (adityaruplaha) [reply electron/protocol.ts:21]: Please restore the mapping changes, it seems dead code, but it'll be useful in the future.

File: public/locales/en/translation.json
Comment (adityaruplaha) [reply public/locales/en/translation.json:119]: Not fixed?

File: electron/protocol.ts
Comment (adityaruplaha) [inline electron/protocol.ts:38]: Hold up, why're we not launching the game from the backend? This introduces an unnecessary dependency on the frontend, which might hurt later since I plan to add options to not lauch the frontend, but instead only the backend. I'll advise leaving a `//TODO` here for future refactors.

File: public/locales/en/translation.json
Comment (Linux4Good) [reply public/locales/en/translation.json]: fixed now

File: src/screens/Settings/index.css
Comment (Linux4Good) [reply src/screens/Settings/index.css]: fixed as well

**Review by adityaruplaha — APPROVED** [review]
LGTM, apart from a couple cosmetic issues.

But, this PR makes a design decision I do not personally agree with, but seems to work fine for now. Check details below.

File: electron/protocol.ts
Comment (Linux4Good) [reply electron/protocol.ts:38]: That's doesn't make sense. There is not launcher that does that.
Both steam and lutris launch themselves. What can we do is to send heroic to the tray.
How can we launch something without user feedback? How can the user know if the game was launched or not?
And if we have an error? Example: authentication expired. The user will be waiting for something that will never happen.

Comment (flavioislima) [reply electron/protocol.ts:38]: I have to agree on that 😛

Comment (flavioislima) [reply electron/protocol.ts:38]: I think we can discuss later on this.
For now we need to merge this to @Nocccer to update the tests for the install functions, etc...

**Comment by adityaruplaha** [comment]
@Linux4Good Congratulations, and thanks for fixing so many long-standing issues with the codebase in one fell swoop.

**Comment by Linux4Good** [comment]
> @Linux4Good Congratulations, and thanks for fixing so many long-standing issues with the codebase in one fell swoop.

Thanks man. I'll try to keep working on Heroic until my vacation ends in July :D

**Pull request merged**
